cmd/otel-allocator/config/config.go

```diff
@@ -42,8 +39,12 @@ const DefaultConfigFilePath string = "/conf/targetallocator.yaml"
 const DefaultCRScrapeInterval model.Duration = model.Duration(time.Second * 30)
 
 type Config struct {
+	ListenAddr             *string
+	KubeConfigFilePath     string
+	ClusterConfig          *rest.Config
+	RootLogger             logr.Logger
```

Comment:
Is it intentional to make these able to be unmarshaled from the config file? If not, I think we want this tag for each line: 
```
`yaml:"-"`
```

Reply:
Good catch. I think the answer is yes for the basic types, and no for the more complex ones.